src/Microsoft.HttpRepl.Telemetry/MacAddressGetter.cs

```diff
@@ -0,0 +1,191 @@
+                if (string.IsNullOrWhiteSpace(shelloutput))
+                {
+                    return null;
+                }
+
+                return ParseMACAddress(shelloutput);
+            }
+            catch (Win32Exception e)
+            {
+                if (e.NativeErrorCode == ErrorFileNotFound)
+                {
+                    return GetMacAddressByNetworkInterface();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+        }
+
+        private static string ParseMACAddress(string shelloutput)
+        {
+            string macAddress = null;
+            foreach (Match match in Regex.Matches(shelloutput, MacRegex, RegexOptions.IgnoreCase))
+            {
+                if (!Regex.IsMatch(match.Value, ZeroRegex))
+                {
+                    macAddress = match.Value;
+                    break;
+                }
+            }
+
+            if (macAddress != null)
+            {
+                return macAddress;
+            }
+            return null;
+        }
+
+        private static string GetIpCommandOutput()
+        {
+            var ipResult = new ProcessStartInfo
+            {
+                FileName = "ip",
+                Arguments = "link",
+                UseShellExecute = false
+            }.ExecuteAndCaptureOutput(out string ipStdOut, out string ipStdErr);
+
+            if (ipResult == 0)
+            {
+                return ipStdOut;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private static string GetShellOutMacAddressOutput()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var result = new ProcessStartInfo
+                {
+                    FileName = "getmac.exe",
+                    UseShellExecute = false
+                }.ExecuteAndCaptureOutput(out string stdOut, out string stdErr);
+
+                if (result == 0)
+                {
+                    return stdOut;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                try
+                {
+                    var ifconfigResult = new ProcessStartInfo
+                    {
+                        FileName = "ifconfig",
+                        Arguments = "-a",
+                        UseShellExecute = false
+                    }.ExecuteAndCaptureOutput(out string ifconfigStdOut, out string ifconfigStdErr);
+
+                    if (ifconfigResult == 0)
+                    {
+                        return ifconfigStdOut;
+                    }
+                    else
+                    {
+                        return GetIpCommandOutput();
+                    }
+                }
+                catch (Win32Exception e)
```

Comment:
I would hope not... but looking back through the history in dotnet/try and dotnet/sdk, its been that way since the beginning, so I'm hesitant to change it right now.